astropy/io/registry/tests/test_registries.py

```diff
@@ -30,6 +31,15 @@
 from astropy.io.registry.compat import default_registry
 from astropy.table import Table
 
+SKIPIF_OPTIMIZED_PYTHON = pytest.mark.skipif(
+    sys.flags.optimize >= 2, reason="docstrings are not available at runtime"
+)
+ONLY_OPTIMIZED_PYTHON = pytest.mark.skipif(
+    sys.flags.optimize < 2,
+    reason="checking behavior specifically if docstrings are not present",
+)
```

Comment:
Should these be generalized and reused elsewhere you did the optimized mode work? We probably don't need such specific reasons.

Reply:
I'm honestly not sure that it'd be worth the cost of less specific reasons + an additional indirection just to avoid repeating two lines over a couple files.
Though, if that's what the majority wants, maybe I could unify the implementation in a cleanup PR when everything is working ?